microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
 
-<!-- ====================================================================== -->
-## Dominators
+The shallow size of an object is usually small because JavaScript objects often just store their own descriptions only in memory. Arrays and strings are exceptions to this and can have a significant shallow size. Most JavaScript objects store their values elsewhere in the renderer memory, and only expose a small wrapper object on the JavaScript heap.
 
-Dominator objects are comprised of a tree structure, because each object has exactly one dominator.  A dominator of an object might lack direct references to an object that it dominates.  That is, dominator's tree is not a spanning tree of the graph.
+Nevertheless, even a small object can hold a large amount of memory indirectly, by preventing other objects from being disposed of by the automatic garbage collection process.
 
-In the following figure:
+<!-- ------------------------------ -->
+#### Retained size
 
-*  Node 1 dominates node 2.
-*  Node 2 dominates nodes 3, 4 and 6.
-*  Node 3 dominates node 5.
-*  Node 5 dominates node 8.
-*  Node 6 dominates node 7.
+The _retained size_ is the size of memory that is implicitly held by an object and that is freed once the object is deleted along with all the dependent objects that were made unreachable from GC roots.
 
-![Dominator tree structure](./memory-101-images/dominatorsspanning.png)
+Said another way, the retained size of an object is the amount of memory that would be regained if the object and all of its dependent objects were removed from the memory graph.
 
-In the following figure, node `#3` is the dominator of node `#10`.  But node `#7` also exists in every simple path from the garbage collection root **GC** to node `#10`. Therefore, an object `B` is a dominator of an object `A` if object `B` exists in every simple path from the root to the object `A`.
+The retained size can't be smaller than the shallow size.
 
-Node `GC` dominates nodes `#1`, `#3`, and `#11`:
 
-![Node GC dominates nodes #1, #3, and #11](./memory-101-images/dominators-00.png)
- 
-Node `#3` is dominated by node `GC` and dominates node `#7`:
+<!-- ------------------------------ -->
+#### Distance
 
-![Node #3 is dominated by node GC and dominates node #7](./memory-101-images/dominators-01.png)
- 
-Node `#7` is dominated by node `#3` and dominates nodes `#8`, `#9`, and `#10`:
+The _distance_ of an object in the JavaScript heap is the number of nodes on the shortest path between the object and the GC root. The shorter the distance, the more likely it is that this object plays an important role in the memory usage of the webpage.
 
-![Node #7 is dominated by node #3 and dominates nodes #8, #9, and #10](./memory-101-images/dominators-02.png)
- 
-Node `#8` is dominated by node `#7` and doesn't dominate any nodes:
+If almost all the objects of the same type are at the same distance, and a few are at a bigger distance, that's something worth investigating.<!-- WHY? -->
```

Comment:
@sethbrenith My understanding is that the `distance` column in the Memory tool isn't super actionable for web developers. This sentence has been in this article since the beginning, and I don't really know what it means. Why would looking at this pattern help web devs?

Reply:
Thanks. Removing the sentence, but adding a sentence about the Retainers section default sorting order.